meson.build

```
@@ -0,0 +1,43 @@
+project('batsignal', 'c',
+  license : 'ISC',
+  version : '1.3.2',
+  default_options : ['c_std=c99']
+)
+
+cc = meson.get_compiler('c')
+os = host_machine.system()
+compat = meson.version().version_compare('<0.1')
+
+libm_args = [] # ['-D', '_POSIX_C_SOURCE=200112L']
+libm = declare_dependency()
+if os == 'freebsd' or \
+    (os == 'linux' and cc.has_header_symbol('features.h', '__GLIBC__'))
+  libm = cc.find_library('m')
+endif
+libm_ok = compat or cc.has_function('round',
+  args: libm_args,
+  dependencies : [libm],
+  prefix : '#include <math.h>')
+assert(libm_ok)
+
+config = configuration_data()
+config.set_quoted('VERSION', meson.project_version())
+configure_file(output : 'config.h',
+  configuration : config)
```

Since meson 0.49 (and compatible with muon too) you can do:
```suggestion
configure_file(output : 'config.h', configuration : {'VERSION': meson.project_version()})
```

---

Your recommended approach does not quote the version. It would generate `1.2.3` instead of `"1.2.3"`.

---

Right, sorry, that was silly of me.